felix/dataplane/linux/bpf_ep_mgr.go

```diff
@@ -4201,10 +4211,246 @@ func (m *bpfEndpointManager) getIfaceTypeFromLink(link netlink.Link) IfaceType {
 }
 
 func (m *bpfEndpointManager) getBondSlaves(masterIfName string) set.Set[string] {
-	if val, ok := m.hostIfaceToSlaveDevices[masterIfName]; ok {
+	slaveDevices := set.New[string]()
+	link, err := m.dp.getIfaceLink(masterIfName)
+	if err != nil {
+		log.Errorf("Error getting link for interface %s, err = %v", masterIfName, err)
+		return slaveDevices
+	}
+	hostIf := m.findHostIfaceByIndex(link.Attrs().Index)
+	if hostIf == nil {
+		log.Errorf("error finding interface %s", masterIfName)
+		return slaveDevices
+	}
+
+	if len(hostIf.children) > 0 {
+		leaves := []string{}
+		getLeafNodes(hostIf, &leaves)
+		slaveDevices.AddAll(leaves)
+	}
+	return slaveDevices
+}
+
+// addHostIface adds host interface to hostIfaces tree.
+func (m *bpfEndpointManager) addHostIface(link netlink.Link) {
+	attrs := link.Attrs()
+	intf := &bpfHostIface{name: attrs.Name,
+		index:       attrs.Index,
+		masterIndex: attrs.MasterIndex,
+		parentIndex: attrs.ParentIndex,
+		children:    make(map[int]*bpfHostIface)}
+
+	if attrs.MasterIndex == 0 && attrs.ParentIndex == 0 {
+		eintf := m.findHostIfaceByIndex(attrs.Index)
+		if eintf != nil {
```

Comment:
I just rewrite the name so that I don't overwrite the children information. For example, when felix gets the child information first, we create the child node and also the parent node with master index. Now when felix gets the update of the parent interface, the interface with that index is already there in the tree. Just need to update the name.